src/Files.Uwp/UserControls/MultitaskingControl/TabItem/TabItem.cs

```diff
@@ -76,27 +97,179 @@ public void Dispose()
         #endregion IDisposable
     }
 
-    public class TabItemArguments
+    public class KnownTypeSerialization
     {
-        private static KnownTypesBinder TypesBinder = new KnownTypesBinder
-        {
-            KnownTypes = new List<Type> { typeof(PaneNavigationArguments) }
-        };
+        private KnownTypesBinder knownTypesBinder = null;
 
-        public Type InitialPageType { get; set; }
-        public object NavigationArg { get; set; }
+        public KnownTypeSerialization()
+        {
+            knownTypesBinder = new KnownTypesBinder
+            {
+                KnownTypes = new List<Type>() { }
+            };
+        }
 
-        public string Serialize() => JsonConvert.SerializeObject(this, new JsonSerializerSettings
+        public string Serialize<T>(T target) => JsonConvert.SerializeObject(target, new JsonSerializerSettings
         {
             TypeNameHandling = TypeNameHandling.Auto,
-            SerializationBinder = TypesBinder
+            SerializationBinder = knownTypesBinder
         });
 
-        public static TabItemArguments Deserialize(string obj) => JsonConvert.DeserializeObject<TabItemArguments>(obj, new JsonSerializerSettings
+        public T Deserialize<T>(string obj) => JsonConvert.DeserializeObject<T>(obj, new JsonSerializerSettings
         {
             TypeNameHandling = TypeNameHandling.Auto,
-            SerializationBinder = TypesBinder
+            SerializationBinder = knownTypesBinder
         });
+
+        public KnownTypeSerialization SetKnownTypes(Type[] types)
+        {
+            knownTypesBinder = new KnownTypesBinder
+            {
+                KnownTypes = types
+            };
+
+            return this;
+        }
+
+        public KnownTypeSerialization SetKnownType(Type type)
+        {
+            knownTypesBinder = new KnownTypesBinder
+            {
+                KnownTypes = new List<Type>() { type }
+            };
+
+            return this;
+        }
+    }
+
+    public class TabItemArguments
+    {
+        public Type InitialPageType { get; set; }
+        public object NavigationArg { get; set; }
+
+        private static Type[] serializationKnownTypes = new Type[]
+        {
+            typeof(PaneNavigationArguments)
+        };
+
+        public string Serialize()
+        {
+            return new KnownTypeSerialization().SetKnownTypes(serializationKnownTypes)
+                                               .Serialize(this);
+        }
+
+        public static TabItemArguments Deserialize(string obj)
+        {
+            return new KnownTypeSerialization().SetKnownTypes(serializationKnownTypes)
+                                               .Deserialize<TabItemArguments>(obj);
+        }
+    }
+
+    public class PropertySerializer<T>
```

Comment:
What's the purpose of it and why is it in `TabItem.cs`?

Reply:
**What's the purpose of it?**
I want to serialize/deserialize only a few properties,  which have the custom attribute `SerializableProperty`, of  `TabItem` objects.

The class `PropertySerializer` is implemented to help with that.

**Why is it in TabItem.cs?**
I thought nowhere else uses it, so I put it here. Now, it seems not good, yeah.

When the implementation is accepted, I will move it to somewhere more general.

Reply:
I moved my custom serialization to the namespace `Files.Helpers` on the latest commit.